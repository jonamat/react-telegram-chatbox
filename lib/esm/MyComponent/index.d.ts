import { FC, ReactNode } from 'react';
export interface MyComponentProps {
    /** Inline prop doc example. Default `world` */
    myProp?: ReactNode;
}
/**
 * MyComponent description.
 *
 * See [Docs](https://github.com/jonamat/react-library-boilerplate).
 */
declare const MyComponent: FC<MyComponentProps>;
export default MyComponent;
