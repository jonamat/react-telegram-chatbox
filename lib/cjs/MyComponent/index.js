"use strict";
/* -------------------------------------------------------------------------- *
 * Copyright (c) Jonathan Mataloni. All rights reserved.
 * Licensed under the MIT License.
 * See License.txt in the project root for license information.
 * -------------------------------------------------------------------------- */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var prop_types_1 = __importDefault(require("prop-types"));
/**
 * MyComponent description.
 *
 * See [Docs](https://github.com/jonamat/react-library-boilerplate).
 */
var MyComponent = function (props) {
    return react_1.default.createElement(react_1.default.Fragment, null,
        "Hello ",
        props.myProp || 'world',
        "!");
};
MyComponent.propTypes = {
    myProp: prop_types_1.default.element,
};
exports.default = MyComponent;
