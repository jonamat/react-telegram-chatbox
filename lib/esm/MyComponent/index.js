/* -------------------------------------------------------------------------- *
 * Copyright (c) Jonathan Mataloni. All rights reserved.
 * Licensed under the MIT License.
 * See License.txt in the project root for license information.
 * -------------------------------------------------------------------------- */
import React from 'react';
import PropTypes from 'prop-types';
/**
 * MyComponent description.
 *
 * See [Docs](https://github.com/jonamat/react-library-boilerplate).
 */
const MyComponent = (props) => {
    return React.createElement(React.Fragment, null,
        "Hello ",
        props.myProp || 'world',
        "!");
};
MyComponent.propTypes = {
    myProp: PropTypes.element,
};
export default MyComponent;
