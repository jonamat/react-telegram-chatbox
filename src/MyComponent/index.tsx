/* -------------------------------------------------------------------------- *
 * Copyright (c) Jonathan Mataloni. All rights reserved.
 * Licensed under the MIT License.
 * See License.txt in the project root for license information.
 * -------------------------------------------------------------------------- */

import React, { FC, ReactNode } from 'react';
import PropTypes from 'prop-types';

export interface MyComponentProps {
    /** Inline prop doc example. Default `world` */
    myProp?: ReactNode;
}

/**
 * MyComponent description.
 *
 * See [Docs](https://github.com/jonamat/react-library-boilerplate).
 */
const MyComponent: FC<MyComponentProps> = (props) => {
    return <>Hello {props.myProp || 'world'}!</>;
};

MyComponent.propTypes = {
    myProp: PropTypes.element,
};

export default MyComponent;
