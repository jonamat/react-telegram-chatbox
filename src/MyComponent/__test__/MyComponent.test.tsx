/* -------------------------------------------------------------------------- *
 * Copyright (c) Jonathan Mataloni. All rights reserved.
 * Licensed under the MIT License.
 * See License.txt in the project root for license information.
 *
 * ----------------------------- Unit test suite ---------------------------- *
 *
 * Target: Database update system (delete data)
 * Tests:
 * - Props validation
 * - Functionality
 * - Matching render snapshot
 *
 * -------------------------------------------------------------------------- */

import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { create } from 'react-test-renderer';

/* ---------------------------- Component to test --------------------------- */

import MyComponent, { MyComponentProps } from '..';

/* -------------------------------------------------------------------------- */
/*                                    Mocks                                   */
/* -------------------------------------------------------------------------- */

const consoleError = jest.fn();
Object.defineProperty(console, 'error', {
    get() {
        return consoleError;
    },
});

/* -------------------------------------------------------------------------- */
/*                              Bridge operations                             */
/* -------------------------------------------------------------------------- */

beforeEach(() => {
    consoleError.mockClear();
});

/* -------------------------------------------------------------------------- */
/*                               Render wrapper                               */
/* -------------------------------------------------------------------------- */

/**
 * Render the component with the custom configuration
 * @param props Component props
 */
const renderComponent = (props?: MyComponentProps) => render(<MyComponent {...props} />);

/* -------------------------------------------------------------------------- */
/*                                    Tests                                   */
/* -------------------------------------------------------------------------- */

describe('Component', () => {
    it('render properly', () => {
        expect(() => {
            const { queryByTestId } = renderComponent({ myProp: <span data-testid="children">mars</span> });

            expect(consoleError).not.toBeCalled();
            expect(queryByTestId('children')).toBeInTheDocument();
        }).not.toThrowError();
    });

    it('validates props', () => {
        expect(() => renderComponent({ myProp: { not: 'a react node' } })).toThrowError();
    });

    it('matches the snapshot', () => {
        const componentSnapshot = create(<MyComponent />).toJSON();
        expect(componentSnapshot).toMatchSnapshot();
    });
});
