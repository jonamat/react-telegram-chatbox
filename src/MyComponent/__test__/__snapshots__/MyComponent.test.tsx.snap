// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches the snapshot 1`] = `
Array [
  "Hello ",
  "world",
  "!",
]
`;
